### PR TITLE
Fix #2973 Added a shared library "libcurl-gnutls.so"

### LINKED ISSUE #2973: contrib/docker: error while loading nginx plugin
*   Version of collectd: 5.7.1
*   Operating system / distribution: debian:stable (stretch 9.5)
*   Docker version: 18.06.1-ce

When building the docker container with [`contrib/docker`](https://github.com/collectd/collectd/tree/master/contrib/docker) source code and loading the nginx plugin, an error occurred at startup and terminated.

## Expected behavior
Running without error.

## Actual behavior
Stopped on error.
```
# docker run -u 0:0 -v $PWD/nginx.conf:/etc/collectd/collectd.conf.d/nginx.conf collectd:5.7.1-1.1
ERROR: lt_dlopen ("/usr/lib/collectd/nginx.so") failed: file not found. The most common cause for this problem is missing dependencies. Use ldd(1) to check the dependencies of the plugin / shared object.
[2018-10-25 04:48:17] [info] plugin_load: plugin "unixsock" successfully loaded.
[2018-10-25 04:48:17] [error] lt_dlopen ("/usr/lib/collectd/nginx.so") failed: file not found. The most common cause for this problem is missing dependencies. Use ldd(1) to check the dependencies of the plugin / shared object.
[2018-10-25 04:48:17] [error] plugin_load: Load plugin "nginx" failed with status 1.
[2018-10-25 04:48:17] [info] plugin_load: plugin "cpu" successfully loaded.
[2018-10-25 04:48:17] [info] plugin_load: plugin "memory" successfully loaded.
[2018-10-25 04:48:17] [info] plugin_load: plugin "disk" successfully loaded.
[2018-10-25 04:48:17] [info] plugin_load: plugin "df" successfully loaded.
Error: Reading the config file failed!
Read the syslog for details.
```
## Steps to reproduce
*   step 1
Clone this repository.
```
# git clone https://github.com/collectd/collectd.git
```
*   step 2
Build containers.
```
# cd collectd/contrib/docker
# docker build -t collectd:5.7.1-1.1 .
```
*   step 3
Edit setting.
```
# echo "LoadPlugin nginx" > nginx.conf
```
*   step 4
Execution.
```
# docker run -u 0:0 -v $PWD/nginx.conf:/etc/collectd/collectd.conf.d/nginx.conf collectd:5.7.1-1.1
```

### PR DESCRIPTION
**Pull request by 3kami3**
> libcurl-gnutls.so.4 => not found

Fix Issues #2973.
Added a shared library "libcurl-gnutls.so".

### CODE REVIEW

File: contrib/docker/Dockerfile [hunk 1]
```diff
@@ -7,9 +7,10 @@ COPY rootfs_prefix/ /usr/src/rootfs_prefix/
 
 RUN apt-get update \
  && apt-get upgrade \
- && apt-get install \
+ && DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
```
Comment (rpv-tomsk) [inline contrib/docker/Dockerfile]: Hi!

1) `DEBIAN_FRONTEND=noninteractive` already set in line 3.
2) `--no-install-recommends` and  `-y` already set in `50docker-apt-conf`.

Please remove these changes.

Thanks for your contribution!

**Comment by rpv-tomsk** [comment]
@octo, @rubenk  - I think `contrib/docker/Dockerfile`  has no goal to allow to build all available plugins and we should decline such PR. Can anyone please confirm or deny this?

File: contrib/docker/Dockerfile (hunk 1 above)
Comment (3kami3) [reply contrib/docker/Dockerfile]: I see.
If you have such a policy, I think it is good to write so.
```diff
--- a/contrib/docker/Dockerfile
+++ b/contrib/docker/Dockerfile
@@ -11,6 +11,8 @@ RUN apt-get update \
     collectd-core \
     collectd-utils \
     build-essential \
+ && : "If you need some shared libraries when enabling plugins please add the package here." \
+ && : "FYI. https://github.com/collectd/collectd/pull/2974" \
  && make -C /usr/src/rootfs_prefix/ \
  && apt-get --purge remove build-essential \
  && apt-get clean \
```

**Comment by rpv-tomsk** [comment]
Ok, will change 'build' to 'run' in my messages.

Debian package has many of supported plugins binaries and also has no strong dependency to allow to run all of them. Dependency is 'soft' and somebody has to install required dependency yourself.

**Comment by octo** [comment]
Setting `--no-install-recommends` is actually part of the problem, not part of the solution. The argument of the Debian maintainers has been that "normal" users will install recommendations by default, so for Joe Average this is as good as a strong dependency.

IMHO, since there's already a complete operating system in the container, we might as well pull in the recommendations.

**Comment by rpv-tomsk** [comment]
Installing Collectd with all recommendations will produce installation of huge amount of packages, most of whose will not be used.

I think we should result on adding a notice, as proposed by @3kami3 comment, and allow to users decide which packages they really need to install.

**Comment by octo** [comment]
The `debian:stable` base is ~100 MByte, all packages depended on or recommended by the `collectd` package amount to 30.8 MByte. Is saving 31 MByte really worth this user frustration? Imho it is more reasonable to have users who care about 31 MByte do the extra work.

Alternatively, we could try using the `debian:stable-slim` as our base (56 MByte), install all recommendations and still have a smaller image.

**Comment by rpv-tomsk** [comment]
@3kami3, can you please check if `APT::Install-Recommends "1";` in `contrib/docker/50docker-apt-conf` produces expected installation of all required dependencies?

Thanks.

**Comment by 3kami3** [comment]
@rpv-tomsk, I checked when `APT::Install-Recommends "1"`.
It seems that all required dependencies are installed.
```
$ git clone https://github.com/collectd/collectd.git
$ cd collectd/contrib/docker
$ git diff
$ docker build -t collectd:master .
Sending build context to Docker daemon  13.82kB
Step 1/9 : FROM debian:stable
 ---> a42749e42c45
Step 2/9 : ENV DEBIAN_FRONTEND noninteractive
 ---> Running in 129df2280bdc
Removing intermediate container 129df2280bdc
 ---> 3ee51fc07412
Step 3/9 : COPY 50docker-apt-conf /etc/apt/apt.conf.d/
 ---> 7c9472d79f00
Step 4/9 : COPY rootfs_prefix/ /usr/src/rootfs_prefix/
 ---> bdd48a25f62e
Step 5/9 : RUN apt-get update  && apt-get upgrade  && apt-get install     collectd-core     collectd-utils     build-essential  && make -C /usr/src/rootfs_prefix/  && apt-get --purge remove build-essential  && apt-get clean  && rm -rf /var/lib/apt/lists/*
 ---> Running in 5dba826654f0
<snip>
Removing intermediate container 5dba826654f0
 ---> 0e053af800d6
Step 6/9 : COPY collectd.conf /etc/collectd/collectd.conf
 ---> 0d32ea0b1e99
Step 7/9 : COPY collectd.conf.d /etc/collectd/collectd.conf.d
 ---> c32eac790ee6
Step 8/9 : ENV LD_PRELOAD /usr/src/rootfs_prefix/rootfs_prefix.so
 ---> Running in d95ea5fe09e3
Removing intermediate container d95ea5fe09e3
 ---> 9d50eb751dad
Step 9/9 : CMD [ "/usr/sbin/collectd", "-f"]
 ---> Running in f871c01f4c9a
Removing intermediate container f871c01f4c9a
 ---> 85d671063828
Successfully built 85d671063828
Successfully tagged collectd:master

$ docker run -it collectd:master bash
root@a6ab16604387:/# cd /usr/lib/collectd/
root@a6ab16604387:/usr/lib/collectd# find . -name "*.so" -exec ldd {} \; | grep "not found"
        librabbitmq.so.4 => not found
        libyajl.so.2 => not found
        libcurl-gnutls.so.4 => not found
<snip>
root@a6ab16604387:/usr/lib/collectd# find . -name "*.so" -exec ldd {} \; | grep "not found" | wc -l
119
root@a6ab16604387:/usr/lib/collectd# exit
exit
```
```
$ sed -i -e 's/Recommends "0"/Recommends "1"/' 50docker-apt-conf
$ git diff
diff --git a/contrib/docker/50docker-apt-conf b/contrib/docker/50docker-apt-conf
index 3f898b3..1b71704 100644
--- a/contrib/docker/50docker-apt-conf
+++ b/contrib/docker/50docker-apt-conf
@@ -1,4 +1,4 @@
-APT::Install-Recommends "0";
+APT::Install-Recommends "1";
 APT::Install-Suggests "0";
 APT::Get::Assume-Yes "1";
 APT::Get::AutomaticRemove "1";

$ docker build -t collectd:full .
Sending build context to Docker daemon  13.82kB
Step 1/9 : FROM debian:stable
 ---> a42749e42c45
Step 2/9 : ENV DEBIAN_FRONTEND noninteractive
 ---> Using cache
 ---> 3ee51fc07412
Step 3/9 : COPY 50docker-apt-conf /etc/apt/apt.conf.d/
 ---> 6ca5dfa0b92b
Step 4/9 : COPY rootfs_prefix/ /usr/src/rootfs_prefix/
 ---> eced9b84af04
Step 5/9 : RUN apt-get update  && apt-get upgrade  && apt-get install     collectd-core     collectd-utils     build-essential  && make -C /usr/src/rootfs_prefix/  && apt-get --purge remove build-essential  && apt-get clean  && rm -rf /var/lib/apt/lists/*
 ---> Running in ecd6e0fb0107
<snip>
Removing intermediate container ecd6e0fb0107
 ---> df463ff2da9a
Step 6/9 : COPY collectd.conf /etc/collectd/collectd.conf
 ---> 054ad26f8ea0
Step 7/9 : COPY collectd.conf.d /etc/collectd/collectd.conf.d
 ---> 4cbb3a9e9ed3
Step 8/9 : ENV LD_PRELOAD /usr/src/rootfs_prefix/rootfs_prefix.so
 ---> Running in 1847a0dc8ba6
Removing intermediate container 1847a0dc8ba6
 ---> 764d6f14d988
Step 9/9 : CMD [ "/usr/sbin/collectd", "-f"]
 ---> Running in 6d3a800ee47e
Removing intermediate container 6d3a800ee47e
 ---> 13c894ffd404
Successfully built 13c894ffd404
Successfully tagged collectd:full

$ docker run -it collectd:full bash
root@f2c38ef0d6fe:/# cd /usr/lib/collectd/
root@f2c38ef0d6fe:/usr/lib/collectd# find . -name "*.so" -exec ldd {} \; | grep "not found"
root@f2c38ef0d6fe:/usr/lib/collectd# find . -name "*.so" -exec ldd {} \; | grep "not found" | wc -l
0
root@f2c38ef0d6fe:/usr/lib/collectd# exit
exit

$ docker images
REPOSITORY                    TAG                 IMAGE ID            CREATED              SIZE
collectd                      full                13c894ffd404        About a minute ago   527MB
collectd                      master              85d671063828        17 minutes ago       148MB
```

**Comment by rpv-tomsk** [comment]
Thanks for working on this issue!

**Pull request closed without merging**